servers/parser-graphql-wrapper/src/context.ts

```diff
@@ -35,25 +39,52 @@ export interface IContext extends ContextWithDataSources {
 export class ContextManager implements IContext {
   public readonly dataLoaders: IContext['dataLoaders'];
   public readonly repositories: IContext['repositories'];
+  public readonly headers: IContext['headers'];
 
   private constructor(
     dataloaders: IContext['dataLoaders'],
     repositories: IContext['repositories'],
+    headers: { [key: string]: any },
   ) {
     this.dataLoaders = dataloaders;
     this.repositories = repositories;
+    this.headers = headers;
   }
-  static async initialize(): Promise<ContextManager> {
+  static async initialize(config: {
+    headers: { [key: string]: any };
+  }): Promise<ContextManager> {
     const dataloaders = {
       itemIdLoader: itemIdLoader,
       itemUrlLoader: itemUrlLoader,
       shortUrlLoader: ShortUrlLoader(),
     };
     const itemResolver = await getItemResolverRepository();
     const sharedUrlsResolver = await getSharedUrlsResolverRepo();
-    return new ContextManager(dataloaders, {
-      itemResolver,
-      sharedUrlsResolver,
-    });
+    return new ContextManager(
+      dataloaders,
+      {
+        itemResolver,
+        sharedUrlsResolver,
+      },
+      config.headers,
+    );
+  }
+
+  get userId(): string | undefined {
```

Comment:
@kschelonka added in this data if it exists so we can target the feature flag by user/ip

Reply:
I'm 100% on board with feature flags!